File: codes/cpp/Code_5_10_2.cpp
#include <iostream>
using namespace std;

int main() {
	long long a, b, c;
	cin >> a >> b >> c;
	if (c - a - b < 0LL) cout << "No" << endl;
	else if (4LL * a * b < (c - a - b) * (c - a - b)) cout << "Yes" << endl;
	else cout << "No" << endl;
	return 0;
}

File: codes/java/Code_5_10_2.java
import java.util.*;

class Code_5_10_2 {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		long a = sc.nextLong();
		long b = sc.nextLong();
		long c = sc.nextLong();
		if (c - a - b < 0) {
			System.out.println("No");
		}
		else if (4 * a * b < (c - a - b) * (c - a - b)) {
			System.out.println("Yes");
		}
		else {
			System.out.println("No");
		}
	}
}
